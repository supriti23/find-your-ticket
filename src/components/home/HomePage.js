import React, { useState } from "react";
import "./HomePage.scss";
import Button from "../button/Button";
import MainForm from "../mainform/MainForm";
const HomePage = () => {
  const [find, setFind] = useState(false);

  function nextClick() {
    setFind(true);
  }
  console.log(find)
  return (
    <>
      {find ? (
        <MainForm />
      ) : (
        <div className="container-fluid home">
          <div className="heading row justify-content-center">
            <p>Welcome to Berlin</p>
          </div>
          <div className="row justify-content-center">
            <Button buttonName="Find Tickets" handleClick={ () => nextClick()}></Button>
          </div>
        </div>
      )}
    </>
  );
};
export default HomePage;
