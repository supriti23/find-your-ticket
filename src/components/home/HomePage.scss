.home {
  background-image: url("../../assets/HomeBackground.jpg");
  background-position: center;
  background-size: 100% auto;
  height: 100vh;
  transform: scale(1);
  background-repeat: no-repeat;
  padding: 10%;
}

.heading {
  font-family: "Courgette", cursive;
  font-size: 8vh;
}

.btn-lg {
  width: 40vh;
  font-family: "Playfair Display", serif;;
}
