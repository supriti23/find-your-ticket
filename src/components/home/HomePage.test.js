import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import HomePage from "./HomePage";
import renderer from "react-test-renderer";

describe("Homepage Testing", () => {
    const component = renderer.create(<HomePage />).toJSON();
  test("Renders without crashing and displays a button with text", () => {
    expect(component).toMatchSnapshot();   
  });

  test("onclick event testing of Find Tickets Button", () => {
    render(<HomePage />)
    const button = screen.getByText("Find Tickets");
    fireEvent.click(button);
  });
});
