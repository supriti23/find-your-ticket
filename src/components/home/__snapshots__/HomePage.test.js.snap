// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Testing Renders without crashing and displays a button with text 1`] = `
<div
  className="container-fluid home"
>
  <div
    className="heading row justify-content-center"
  >
    <p>
      Welcome to Berlin
    </p>
  </div>
  <div
    className="row justify-content-center"
  >
    <button
      className="btn btn-primary btn-lg"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
      type="button"
    >
      Find Tickets
    </button>
  </div>
</div>
`;
